### ISSUE TITLE
Wanted: extend `@stylistic/curly-newline` for code in templates

### ISSUE THREAD

**Issue by justingrant** [vuejs/eslint-plugin-vue#2614]
**Please describe what the rule should do:**

Require (or disallow) line breaks after and before curly brackets used in attribute values and inside `{{ }}` expressions. (Although not for the handlebars themselves!)

There is already a similar rule for square brackets for arrays: [vue/array-bracket-newline](https://eslint.vuejs.org/rules/array-bracket-newline.html). But the corresponding rule for curly brackets is missing.

The [`@stylistic/curly-newline`](https://eslint.style/rules/default/curly-newline) rule already exists for this purpose, and it's is helpful for standardizing how multi-line blocks are broken across lines. In Vue templates, this is most commonly needed in style objects or property-bag parameters to functions.

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
 <div
    v-if="$vuetify.breakpoint.smAndDown"
    :class="{ active: isActive,
              inactive: !isActive }"
    :style="bgStyle"
  >
```

```vue
 <img
    v-if="item.thumbnails"
    :style="{ 'max-width': `${thumbnailSize}px`,
              'max-height': `${thumbnailSize}px` }"
    :src="foo.png"
  >
```

```vue
          {{
            someFunc({ opt1: 'true',
                       opt2: 'allow' })
          }}
```

**Additional context**

**Comment by waynzh** [comment]
Perhaps you could try [vue/object-curly-newline](https://eslint.vuejs.org/rules/object-curly-newline.html) to see if it meets your needs? 
And refer to [@stylistic/object-curly-newline](https://eslint.style/rules/default/object-curly-newline) for configuration options.

**Comment by FloEdelmann** [comment]
@waynzh is right: [@stylistic/curly-newline](https://eslint.style/rules/default/curly-newline) is for `if`/`for` and other control statements (which can't be used in Vue files' `<template>` part), while [@stylistic/object-curly-newline](https://eslint.style/rules/default/object-curly-newline) is for objects (and available in eslint-plugin-vue as [vue/object-curly-newline](https://eslint.vuejs.org/rules/object-curly-newline.html). So I'm closing this as answered and won't fix.

**Comment by justingrant** [comment]
Oh great, sorry I didn't see that rule. I'll try it out. Thanks!
